packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -0,0 +1,572 @@
+     * @param opts.recipient Optional address to receive the funds on L1. Defaults to sender.
+     * @returns Transaction response for the withdraw transaction.
+     */
+    public async withdrawERC20(
+        l1Token: AddressLike,
+        l2Token: AddressLike,
+        amount: NumberLike,
+        opts?: {
+            recipient?: AddressLike
+            signer?: Signer
+        }
+    ): Promise<TransactionResponse> {
+        return (opts?.signer || this.l2Signer).sendTransaction(
+            await this.populateTransaction.withdrawERC20(
+                l1Token,
+                l2Token,
+                amount,
+                opts
+            )
+        )
+    }
+
+    /**
+    * Finalizes the deposit on the L2 chain.
+    *
+    * @param bridgeTxLogs Deposit transaction receipt
+    * @param opts Additional options.
+    * @param opts.signer Optional signer to use to send the transaction.
+    * @param opts.l2GasLimit Optional gas limit to use for the transaction on L2.
+    * 
+    * @returns Transaction response for the finalizeDeposit transaction.
+    */
+    public async finalizeDeposit(
+        bridgeTxLogs: TransactionReceipt,
+        opts?: {
+            signer?: Signer
+            l2GasLimit?: NumberLike
+        }): Promise<TransactionResponse> {
+
+        return (opts?.signer || this.l1Signer).sendTransaction(
+            await this.populateTransaction.finalizeDeposit(bridgeTxLogs)
+        )
+
+    }
+
+    /** 
+     * Finalizes the withdrawal on the L1 chain.
+     */
+    // public async finalizeWithdrawal(
+    //     bridgeTxLogs: TransactionReceipt,
+    //     opts?: {
+    //         signer?: Signer
+    //         l2GasLimit?: NumberLike
+    //     }): Promise<TransactionResponse> {
+
+    //     return (opts?.signer || this.l1Signer).sendTransaction(
+    //         await this.populateTransaction.finalizeWithdrawal(bridgeTxLogs)
+    //     )
+    // }
+
+    /**
+     * Object that holds the functions that generate transactions to be signed by the user.
+     * Follows the pattern used by ethers.js.
+     */
+    populateTransaction = {
+
+        /**
+         * Generates a transaction for depositing some ETH into the L2 chain.
+         *
+         * @param amount Amount of ETH to deposit.
+         * @param opts Additional options.
+         * @param opts.recipient Optional address to receive the funds on L2. Defaults to sender.
+         * @param opts.l2GasLimit Optional gas limit to use for the transaction on L2.
+         * @returns Transaction that can be signed and executed to deposit the ETH.
+         */
+        depositETH: async (
+            amount: NumberLike,
+            opts?: {
+                recipient?: AddressLike
+                l2GasLimit?: NumberLike
+            },
+        ): Promise<TransactionRequest> => {
+
+            const bridgeTx = await this.l1StandardBridge.populateTransaction.bridgeETH(200_000, [], {
```

Comment:
i know `200_000` is taken from the e2e test, but it's a random amount.
maybe we can try a better default value? also imo users should be able to overwrite this